content/sql/concepts/window-functions/terms/lag/lag.md

```diff
@@ -59,3 +59,41 @@ Output
 | jenna      | black     | 35  | 60           |
 | chris      | smith     | 30  | 35           |
 | dave       | james     | 19  | 30           |
+
+## Example 
+### Using partition by
+
+This example demonstrates how to use the `LAG()` function to create a new column, `previous_position`. The function retrieves the value from the previous row within the same `employee_id` group in the `Promotions` table. This is achieved by using the `PARTITION BY employee_id` clause.
+
+`Promotions` Table
+
+| employee_id | promotion_date | new_position |
+| ----------- | -------------- | ------------ |
+| 1           | 2020-01-01     | Junior Dev   |
+| 1           | 2021-06-01     |    Mid Dev   |
+| 1           | 2024-03-01     | Senior Dev   |
+| 2           | 2019-05-01     |     Intern   |
+| 2           | 2022-11-01     |    Analyst   |
+| 2           | 2024-11-20     |Data Analyst  |
+
+```sql
+SELECT *,
+    LAG(new_position) OVER (
+        PARTITION BY employee_id
+        ORDER BY promotion_date
+    ) AS previous_position
+FROM Promotions;
+```
+
+The output is a table that features a new column `previous_position`, which holds the values by employee. The first record is null because a default was not specified and the previous row would be out of range.
```

Comment:
The explanation in the second example ("The first record is null because a default was not specified...") is almost identical to the explanation in the first example. We should rewrite it as follows:


```suggestion
Within each group defined by `employee_id`, the `previous_position` column holds the value from the previous row based on `promotion_date`. The first record in each group is `NULL` because there is no preceding row.
```